src/options.rs

```diff
@@ -47,28 +51,63 @@ impl Options {
         }
     }
 
-    pub fn setter_enabled(&self) -> bool {
-        self.setter_enabled
+    pub fn setter_prefix(&self) -> &str {
+        &self.setter_prefix
+    }
+
+    pub fn getter_enabled(&self) -> bool {
+        self.getter_enabled
+    }
+
+    pub fn _getter_visibility(&self) -> Option<quote::Tokens> {
+        if self.getter_public {
+            Some(quote!(pub))
+        } else {
+            None
+        }
+    }
+
+    pub fn _getter_prefix(&self) -> &str {
+        &self.getter_prefix
     }
 }
 
-impl<T> From<T> for Options where
-    T: IntoIterator<Item=syn::Attribute>
+impl<'a, T> From<T> for Options where
+    T: IntoIterator<Item=&'a syn::Attribute>
 {
     fn from(attributes: T) -> Self {
         trace!("Parsing struct attributes.");
-        let mut ob = OptionsBuilder::default();
-
-        for attr in attributes {
-            ob.parse_attribute(attr);
-        }
+        let mut builder = OptionsBuilder::<StructMode>::default();
+        builder.parse_attributes(attributes);
 
-        ob.into()
+        builder.into()
     }
 }
 
+pub trait OptionsBuilderMode {
+    // associated consts are not a thing yet :-/
+    // https://github.com/rust-lang/rust/issues/29646
```

Comment:
I think making it less complex is a good idea. We can always overengineer it later 😆